spec/classes/agent_upgrade_spec.rb

```diff
@@ -4,47 +4,42 @@
   context 'supported operating systems' do
     on_supported_os.each do |os, facts|
       context "on #{os}" do
-        [true, false].each do |mco|
-          context "with mcollective_configured = #{mco}" do
-            let(:facts) do
-              facts.merge({
-                :puppet_ssldir   => '/dev/null/ssl',
-                :puppet_config   => '/dev/null/puppet.conf',
-                :mcollective_configured => mco,
-              })
-            end
+        let(:facts) do
+          facts.merge({
+            :puppet_ssldir   => '/dev/null/ssl',
+            :puppet_config   => '/dev/null/puppet.conf',
+          })
+        end
 
-            if Puppet.version < "3.8.0"
-              it { expect { is_expected.to contain_package('agent_upgrade') }.to raise_error(Puppet::Error, /upgrading requires Puppet 3.8/) }
-            else
-              [{}, {:service_names => []}].each do |params|
-                context "agent_upgrade class without any parameters" do
-                  let(:params) { params }
+        if Puppet.version < "3.8.0"
+          it { expect { is_expected.to contain_package('agent_upgrade') }.to raise_error(Puppet::Error, /upgrading requires Puppet 3.8/) }
+        else
+          [{}, {:service_names => []}].each do |params|
+            context "agent_upgrade class without any parameters" do
+              let(:params) { params }
```

Comment:
It seems the value of this affects whether the compiled catalog contains service resources, but I didn't understand how that works, some rspec-puppet magic?